Documentation/teaching/labs/device_drivers.rst

```diff
@@ -561,17 +561,12 @@ The structure of the write function is similar:
 		       size_t size, loff_t * offset)
    {
        struct my_device_data *my_data = (struct my_device_data *) file->private_data;
-       ssize_t len = min(my_data->size - *offset, size);
-
-       if (len <= 0)
-	   return 0;
 
        /* read data from device in my_data->buffer */
-       if (copy_to_user(user_buffer, my_data->buffer, len))
+       if (copy_from_user(my_data->buffer, user_buffer, my_data->size))
 	   return -EFAULT;
-
-       *offset += len;
-       return lent;
+	   
```

Comment:
Hi Robert,

This is a very good catch. Anyhow, I would like you to do some changes:

1) The commit message should be more clear.

You can use:

Documentation: teaching: labs: Fix data transfer

At write we need to get data from user and write them to device, so use copy_from_user.

2) Also update the comment, it should say /* write data from my_data->buffer to device.

Reply:
Thank you for your speedy reply, I will adjust my commit.
From now on please use my name. 
My name is Teo not Robert

Reply:
Also I used copy_from_user, I do not see where the problem lies.

Reply:
Hi Teo,

Sorry! I was replying to two pull requests in the same time :)